server/lib/chat/chat-service.ts

```diff
@@ -765,18 +822,30 @@ export default class ChatService {
     return userSockets
   }
 
-  private subscribeUserToChannel(
-    userSockets: UserSocketsGroup,
-    channelInfo: ChannelInfo,
-    channelPermissions: ChannelPermissions,
-  ) {
-    userSockets.subscribe<ChatInitEvent>(getChannelPath(channelInfo.id), () => ({
-      action: 'init3',
-      channelInfo,
-      activeUserIds: this.state.channels.get(channelInfo.id)!.toArray(),
-      selfPermissions: channelPermissions,
-    }))
-    userSockets.subscribe(getChannelUserPath(channelInfo.id, userSockets.userId))
+  private subscribeUserToChannel(userSockets: UserSocketsGroup, channelId: SbChannelId) {
+    userSockets.subscribe<ChatInitEvent>(getChannelPath(channelId), async () => {
+      const [channelInfo, userChannelEntry] = await Promise.all([
+        getChannelInfo(channelId),
+        getUserChannelEntryForUser(userSockets.userId, channelId),
+      ])
+      if (!channelInfo) {
+        throw new ChatServiceError(ChatServiceErrorCode.ChannelNotFound, 'Channel not found')
```

Comment:
If you're ever throwing an Error you should ask yourself: What am I expecting to catch this? There's nothing to catch this, you should not be throwing in here.

If you'd like to log something, sure, go ahead. But please do not throw.

Reply:
Relatedly, you probably want to wrap this in a try/catch and log the error should one occur (returning undefined in that case is fine, it will not send anything to the client then)